### PR TITLE
Add basic tests for `PartAlias`, `PartCache`, and matrix truncation

### PR DESCRIPTION
**Pull request by raon0211**
I added some basic tests for `PartAlias`, `PartCache`, and matrix truncation logic.

### CODE REVIEW

File: ldraw/src/library.rs [hunk 1]
```diff
@@ -381,3 +381,46 @@ where
...
+            },
+            subparts: HashMap::new(),
+        };
+
+        let mut cache = PartCache::new();
+
+        let existing_key = PartAlias::from("existing".to_string());
+        let document = Arc::new(document);
+
+        cache.register(PartKind::Primitive, existing_key.clone(), Arc::clone(&document));
```
Comment (kiwiyou) [inline ldraw/src/library.rs:411]: You can use `document.clone()` instead of `Arc::clone(&document)`

Comment (raon0211) [reply ldraw/src/library.rs]: Thanks! Applied in 2e2b213 👍

File: renderer/src/utils.rs [hunk 2]
```diff
@@ -23,3 +23,53 @@ pub(crate) fn derive_normal_matrix(m: &Matrix4) -> Matrix3 {
...
+            Matrix3::new(
+                1.0, 2.0, 3.0,
+                5.0, 6.0, 7.0,
+                9.0, 10.0, 11.0
+            )
+        )
+    }
+
+    #[test]
+    fn test_derive_normal_matrix_1() {
```
Comment (segfault87) [inline renderer/src/utils.rs:56]: This name implies that there should be another subsequent tests on `derive_normal_matrix` (`test_derive_normal_matrix_2`, ...) Is that intentional?

Comment (raon0211) [reply renderer/src/utils.rs]: I originally intended to do so, but it seems that a single test is sufficient 😓  I have edited the test name in dba6e30. Thanks! 👍

**Review by segfault87 — APPROVED** [review]
:100:

**Review by kiwiyou — APPROVED** [review]

**Pull request merged**